syntax = "proto2";

package CitbrainsMessage;
option optimize_for = CODE_SIZE; //これどうなんだっけわすれた

message Pos2D {
  required sint32 pos_x = 1;  //intにして詰める。100倍して100で割る(暫定)
  required sint32 pos_y = 2;
  required sint32 pos_th = 3;
}

message Pos2DCf {
  // optional bytes robot_side = 1;    //TODO こんなん使ってたっけ？
  required Pos2D position = 1;
  optional bytes flags = 2; //TODO フラグ類を詰め込むやつにしたい。Pos2Dをもつ4つのうちどれなのか判断
  optional float cf = 3;
  optional sint32 time = 4;
  /*
  フラグの中身
  robot_side 使ってなさそう
  if_detect 使ってなさそう
  多分enum使うと綺麗になるな。実装と分離するとかいうキモい事が無くなると思う。
  */
}


message SharingData {
  optional bytes id  = 1 ;  //TODO:これはboolとどちらが性能が良いか
  optional bytes cf_own = 2;
  optional bytes cf_ball = 3;
  optional bytes status = 4;
  optional bytes fps = 5;
  optional bytes voltage = 6;     //今は使ってない。
  optional bytes temperature = 7;
  optional bytes highest_servo = 8;
  optional bytes command = 9;
  optional bytes current_behavior_name = 10;
  
  optional bool is_detect_ball = 11;

  repeated Object our_robot_gl = 12; 
  repeated Object enemy_robot_gl = 13; 
  repeated Object black_pole_gl = 14; 
  repeated Object target_pos_vec = 15; 
}

